extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -220,9 +221,19 @@ public void run()
     }
     finally {
       runFuture.set(null);
+      reportCpuMetrics();
     }
   }
 
+  private void reportCpuMetrics()
+  {
+    long cpuTimeNs = 0L;
+    for (final CounterTracker tracker : stageCounters.values()) {
+      cpuTimeNs += tracker.totalCpu();
+    }
+    context.emitMetric("query/cpu/time", TimeUnit.NANOSECONDS.toMicros(cpuTimeNs));
```

Comment:
This one should also have all the same metric dimensions that `query/time` has.

Reply:
Changed.
I assume that for worker metrics, would the datasource, interval, duration etc be based on the stages for which it has actually received work orders. Is that right?

Reply:
For ease of correlating metrics across workers, let's have all workers compute these dimensions based on the overall query, even if they didn't process all stages. Later on we will want to add stage-specific metrics with stage-specific dimensions.